### PR TITLE
Add `CODEOWNERS`

### PR DESCRIPTION
**Pull request by jthodge**
This PR adds a `CODEOWNERS` file to automatically request reviews from the associated and responsible WorkOS team members.

### CODE REVIEW

**Review by bukinoshita — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# See GitHub's docs for more details:
+# https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners
+
+# TypeScript Team
+* @workos-inc/typescript
```
Comment (zlwaterfield) [inline .github/CODEOWNERS:5]: This is is public repo but I don't think our teams are public: https://github.com/orgs/workos-inc/teams/typescript

Comment (maxdeviant) [reply .github/CODEOWNERS:5]: Do the teams need to be public for `CODEOWNERS` to work?

Comment (marktran) [reply .github/CODEOWNERS:5]: I don't believe it does. Seems to register correctly:
> bukinoshita approved these changes on behalf of workos-inc/typescript 6 minutes ago

Comment (maxdeviant) [reply .github/CODEOWNERS:5]: Yea, I think this is fine.

GitHub is still aware of the private team even if you aren't part of the @workos-inc org (this is how it looks if you're not signed into GitHub):

Screen Shot 2020-11-17 at 3 47 51 PM

**Review by maxdeviant — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (jthodge) [reply .github/CODEOWNERS:5]: Yep! Thanks for the questions and responses, y'all. Going to merge :shipit:

**Pull request merged**